### PR TITLE
Add ability to specify Client capabilities

### PR DESCRIPTION
**Pull request by tigrannajaryan**
Resolves https://github.com/open-telemetry/opamp-go/issues/98

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
@andykellr please review.

File: client/internal/clientcommon.go [hunk 1]
```diff
@@ -62,6 +69,9 @@ func (c *ClientCommon) PrepareStart(
 		return errAlreadyStarted
 	}
 
+	c.Capabilities = settings.Capabilities
+	c.Capabilities |= protobufs.AgentCapabilities_ReportsStatus
```
Comment (andykellr) [inline client/internal/clientcommon.go:73]:
```suggestion
        // all agents are required to ReportStatus
	c.Capabilities |= protobufs.AgentCapabilities_ReportsStatus
```

File: client/internal/receivedprocessor.go [hunk 2]
```diff
@@ -57,26 +62,63 @@ func (r *receivedProcessor) ProcessReceivedMessage(ctx context.Context, msg *pro
 			r.logger.Errorf("cannot processed received flags:%v", err)
 		}
 
-		msgData := &types.MessageData{
-			RemoteConfig: msg.RemoteConfig,
+		msgData := &types.MessageData{}
+
+		if msg.RemoteConfig != nil {
+			if r.capabilities&protobufs.AgentCapabilities_AcceptsRemoteConfig != 0 {
```
Comment (andykellr) [inline client/internal/receivedprocessor.go:68]: This is really minor, but I find these hard to read especially with the decision in go to omit space on either side of the ampersand. I would refactor to a function:

```go
func (r *receivedProcessor) hasCapability(capability AgentCapabilities) {
    return r.capabilities&capability != 0
}
```

And then use that throughout.

**Review by andykellr — APPROVED** [review]

File: client/internal/receivedprocessor.go (hunk 2 above)
Comment (tigrannajaryan) [reply client/internal/receivedprocessor.go]: I like it. Done.

**Comment by tigrannajaryan** [comment]
This is now rebased and uses the new ReportsHealth capability as well.

File: client/internal/clientcommon.go (hunk 1 above)
Comment (tigrannajaryan) [reply client/internal/clientcommon.go:75]: Done.

**Pull request merged**